src/topic/BecomeMemberContainer.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import TopicModel from '../shared/models/TopicModel';
+import { becomeMember, leaveTopic } from '../actions/member.actions';
+import { UserModel } from '../shared/models/UserModel';
+
+class BecomeMemberContainer extends React.Component {
+    render() {
+        const { topic, user, attend, leave } = this.props;
+        const alreadyAttending = topic.amIAttending(user);
+        return (
+            <div>
+                {alreadyAttending && (
+                    <button
+                        style={leaveTopicButtonStyle}
+                        onClick={() => {
+                            leave(topic, user);
+                        }}
+                    >
+                        <span style={becomeMemberTextStyle}>Rezygnuję</span>
+                    </button>
+                )}
+                {!alreadyAttending && (
+                    <button
+                        style={becomeMemberButtonStyle}
+                        onClick={() => {
+                            attend(topic, user);
+                        }}
+                    >
+                        <span style={becomeMemberTextStyle}>
+                            Chcę być uczestnikiem
+                        </span>
+                    </button>
+                )}
+            </div>
+        );
+    }
+}
+
+const mapStateToProps = state => ({
+    user: state.auth.user,
+});
+
+const mapDispatchToProps = dispatch => ({
+    attend: (topic, user) => dispatch(becomeMember(topic, user)),
+    leave: (topic, user) => dispatch(leaveTopic(topic, user)),
+});
+
+BecomeMemberContainer.propTypes = {
+    attend: PropTypes.func.isRequired,
+    leave: PropTypes.func.isRequired,
+    user: PropTypes.instanceOf(UserModel).isRequired,
+    topic: PropTypes.instanceOf(TopicModel).isRequired,
+};
+
+export default connect(
+    mapStateToProps,
+    mapDispatchToProps
+)(BecomeMemberContainer);
+
+const actionButtonStyle = {
+    width: '201px',
```

Comment:
please change to styled components way

Reply:
changed in https://github.com/warsawjs/topics-manager/commit/6232648170503c80518ff8059c6d3be5975f81a1